static_build.sh

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env bash
+
+set -ex
+export GO111MODULE="on"
+export GOFLAGS="-mod=vendor"
+export CGO_ENABLED=1
+
+setup_vendor(){
+  LIBGIT2_URL=https://github.com/libgit2/libgit2.git
+  LIBGIT2_VER=v0.28.2
+
+  go mod download
+
+  if ! [[ -d vendor ]]; then
+    go mod vendor
+  fi
+
+  git clone --depth 1 -b $LIBGIT2_VER $LIBGIT2_URL $LIBGIT2_PATH || :
+}
+
+build_libgit2_linux(){
+  cmake \
+  -G Ninja \
+  -DTHREADSAFE=ON \
+  -DBUILD_CLAR=OFF \
+  -DBUILD_SHARED_LIBS=OFF \
+  -DUSE_EXT_HTTP_PARSER=OFF \
+  -DUSE_BUNDLED_ZLIB=ON \
+  -DUSE_HTTPS=OFF \
+  -DUSE_SSH=OFF \
+  -DUSE_ICONV=OFF \
+  -DCMAKE_SYSTEM_NAME=Linux \
+  -DCMAKE_C_COMPILER=${CC} \
+  -DCMAKE_C_FLAGS=-fPIE \
+  -DCMAKE_BUILD_TYPE="RelWithDebInfo" \
+  -DCMAKE_INSTALL_PREFIX="${LIBGIT2_STATIC_PREFIX}" \
+  ${LIBGIT2_PATH}
+}
+
+build_libgit2_windows(){
+  cmake \
+  -G Ninja \
+  -DTHREADSAFE=ON \
+  -DBUILD_CLAR=OFF \
+  -DBUILD_SHARED_LIBS=OFF \
+  -DUSE_EXT_HTTP_PARSER=OFF \
+  -DUSE_BUNDLED_ZLIB=ON \
+  -DUSE_HTTPS=OFF \
+  -DUSE_SSH=OFF \
+  -DUSE_ICONV=OFF \
+  -DWINHTTP=OFF \
+  -DCMAKE_SYSTEM_NAME=Windows \
+  -DCMAKE_C_COMPILER=${CC} \
+  -DCMAKE_BUILD_TYPE="RelWithDebInfo" \
+  -DCMAKE_INSTALL_PREFIX="${LIBGIT2_STATIC_PREFIX}" \
+  -DWIN32=ON \
+  -DMINGW=ON \
+  -DCMAKE_SIZEOF_VOID_P=8 \
+  ${LIBGIT2_PATH}
+}
+
+build_libgit2_darwin(){
+  cmake \
+  -G "Ninja" \
+  -DTHREADSAFE=ON \
+  -DBUILD_CLAR=OFF \
+  -DBUILD_SHARED_LIBS=OFF \
+  -DUSE_EXT_HTTP_PARSER=OFF \
+  -DUSE_BUNDLED_ZLIB=ON \
+  -DUSE_HTTPS=OFF \
+  -DUSE_SSH=OFF \
+  -DUSE_ICONV=OFF \
+  -DCMAKE_SYSTEM_NAME=Darwin \
+  -DCMAKE_OSX_DEPLOYMENT_TARGET="10.14" \
+  -DCMAKE_C_COMPILER=${CC} \
+  -DCMAKE_BUILD_TYPE="RelWithDebInfo" \
+  -DCMAKE_INSTALL_PREFIX="${LIBGIT2_STATIC_PREFIX}" \
+  -DCMAKE_OSX_SYSROOT="${OSX_SDK}" \
+  ${LIBGIT2_PATH}
+}
+
+build_libgit2(){
+  mkdir -vp $LIBGIT2_BUILD
+  pushd $LIBGIT2_BUILD
+
+  case "$TARGET_OS_ARCH" in
+  linux/amd64*)
+    export GOOS=linux GOARCH=amd64 CC=clang
+    FLAGS=""
+    export CGO_LDFLAGS="${LIBGIT2_STATIC_PREFIX}/lib/libgit2.a -L${LIBGIT2_STATIC_PREFIX}/include ${FLAGS}"
+    build_libgit2_linux
+  ;;
+  darwin/amd64*)
+    export GOOS=darwin GOARCH=amd64 CC=x86_64-apple-darwin18-clang OSX_SDK=/opt/osxcross/SDK/MacOSX10.14.sdk/
+    FLAGS=""
+    export CGO_LDFLAGS="${LIBGIT2_STATIC_PREFIX}/lib/libgit2.a -L${LIBGIT2_STATIC_PREFIX}/include ${FLAGS}"
+    build_libgit2_darwin
+  ;;
+  windows/amd64*)
+    export GOOS=windows GOARCH=amd64 CC=x86_64-w64-mingw32-clang
+    FLAGS="-lws2_32"
+    export CGO_LDFLAGS="${LIBGIT2_STATIC_PREFIX}/lib/libgit2.a -L${LIBGIT2_STATIC_PREFIX}/include ${FLAGS}"
+    build_libgit2_windows
+  ;;
+  windows/386*)
+    export GOOS=windows GOARCH=386 CC=i686-w64-mingw32-clang
+    FLAGS="-lws2_32"
+    export CGO_LDFLAGS="${LIBGIT2_STATIC_PREFIX}/lib/libgit2.a -L${LIBGIT2_STATIC_PREFIX}/include ${FLAGS}"
+    build_libgit2_windows
+  ;;
+  esac
+
+  cmake --build . -- -j$(nproc)
```

Comment:
It seems that `nproc` doesn't exists for mac os. Should we consider the static compilation flow just for linux for now? I saw your point at line 123.

Reply:
Yes, static_build.sh assumes your host is linux/amd64. If you want to run the script on macOS, I would suggest you run it in a docker container:
```
$ docker run -it -v $PWD:$PWD -w $PWD navigaid/osxcross ./static_build.sh darwin/amd64
```

Reply:
fixed in https://github.com/cloudson/gitql/pull/82/commits/8f9dda852fe02cb209c7052f9f5dc5efa0aa3b7a
it now uses `-j$(nproc 2>/dev/null || sysctl -n hw.ncpu)`